### PR TITLE
[vcpkg/scripts] add variable VCPKG_CROSSCOMPILING

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

File: scripts/cmake/vcpkg_common_definitions.cmake [hunk 1]
```diff
@@ -22,13 +22,19 @@ TARGET_TRIPLET                           the name of the current triplet to buil
 CURRENT_INSTALLED_DIR                    the absolute path to the installed files for the current triplet
 HOST_TRIPLET                             the name of the triplet corresponding to the host
 CURRENT_HOST_INSTALLED_DIR               the absolute path to the installed files for the host triplet
+VCPKG_CROSSCOMPILING                     ON if TARGET_TRIPLET != HOST_TRIPLET
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_common_definitions.cmake:25]:
```suggestion
VCPKG_IS_CROSSCOMPILING                  ON if TARGET_TRIPLET != HOST_TRIPLET
```

File: scripts/cmake/vcpkg_common_definitions.cmake [hunk 2]
```diff
@@ -22,13 +22,19 @@ TARGET_TRIPLET                           the name of the current triplet to buil
...
 CMAKE_STATIC_LIBRARY_(PREFIX|SUFFIX), CMAKE_SHARED_LIBRARY_(PREFIX|SUFFIX) and CMAKE_IMPORT_LIBRARY_(PREFIX|SUFFIX) are defined for the target
 Furthermore the variables CMAKE_FIND_LIBRARY_(PREFIXES|SUFFIXES) are also defined for the target so that
 portfiles are able to use find_library calls to discover dependent libraries within the current triplet for ports.
 #]===]
 
+set(VCPKG_CROSSCOMPILING OFF)
+if(NOT TARGET_TRIPLET STREQUAL HOST_TRIPLET)
+    set(VCPKG_CROSSCOMPILING ON)
+endif()
+
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_common_definitions.cmake:37]:
```suggestion
string(COMPARE EQUAL "${TARGET_TRIPLET}" "${HOST_TRIPLET}" VCPKG_IS_CROSSCOMPILING)
```

**Review by strega-nil — CHANGES_REQUESTED** [review]
The variable should be called "VCPKG_IS_CROSSCOMPILING"

**Comment by strega-nil** [comment]
Marking requires:discussion since this is an auto-stable new name.

**Comment by strega-nil** [comment]
@Neumann-A please merge the changes; `VCPKG_CROSSCOMPILING` has been agreed upon (2F, 1A)

**Comment by Neumann-A** [comment]
@strega-nil done

**Review by strega-nil — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks!

File: scripts/cmake/vcpkg_common_definitions.cmake (hunk 2 above)
Comment (autoantwort) [reply scripts/cmake/vcpkg_common_definitions.cmake]: `COMPARE EQUAL` ?? 🧐  
I think we are cross compiling when ${TARGET_TRIPLET} is not equal to ${HOST_TRIPLET}

I also have reviewed the PR, but now wonder why changes that relaying on VCPKG_CROSSCOMPILING does not work as expected :D

**Pull request merged**